pkg/fanal/artifact/local/fs.go

```diff
@@ -51,24 +52,41 @@ func NewArtifact(rootPath string, c cache.ArtifactCache, opt artifact.Option) (a
 	return Artifact{
 		rootPath:       filepath.Clean(rootPath),
 		cache:          c,
-		walker:         walker.NewFS(buildAbsPaths(rootPath, opt.SkipFiles), buildAbsPaths(rootPath, opt.SkipDirs), opt.Slow),
+		walker:         walker.NewFS(buildPathsToSkip(rootPath, opt.SkipFiles), buildPathsToSkip(rootPath, opt.SkipDirs), opt.Slow),
 		analyzer:       a,
 		handlerManager: handlerManager,
 
 		artifactOption: opt,
 	}, nil
 }
 
-func buildAbsPaths(base string, paths []string) []string {
-	var absPaths []string
+// buildPathsToSkip builds correct patch for skipDirs and skipFiles
+func buildPathsToSkip(base string, paths []string) []string {
+	var relativePaths []string
+	absBase, err := filepath.Abs(base)
+	if err != nil {
+		log.Logger.Warnf("can't get abspath to base to: %s", err)
+		return nil
+	}
 	for _, path := range paths {
 		if filepath.IsAbs(path) {
-			absPaths = append(absPaths, path)
+			relPath, err := filepath.Rel(absBase, path)
+			if err != nil {
+				log.Logger.Warnf("can't get relative path to %s: %s", path, err)
+				continue
+			}
+			relativePaths = append(relativePaths, filepath.Join(base, relPath))
 		} else {
-			absPaths = append(absPaths, filepath.Join(base, path))
+			// we use not only path: https://github.com/aquasecurity/trivy/blob/8569d43a7a4c0b7301f90c004e7a3b75c0b02bcf/pkg/fanal/walker/fs.go#L33
+			// that is why we need to join base to path
+			relPath := filepath.Join(base, path)
+			if rel, err := filepath.Rel(base, path); err == nil {
+				relPath = filepath.Join(base, rel)
+			}
```

Comment:
We should leave a comment here about why we need this logic.

```suggestion
			// Supports two types of flag specification
			// 1. Relative skip dirs/files from the root directory
			//     The specified dirs and files must be joined with the root directory.
			//       e.g. $ trivy fs --skip-dirs bar ./foo
			//     The path from the working directory will be `foo/bar`.
			// 2. Relative skip dirs/files from the working directory
			//     The specified dirs and files must not be joined with the root directory.
			//       e.g. $ trivy fs --skip-dirs ./foo/bar ./foo

			rel, err := filepath.Rel(base, path); err != nil {
				log.Logger.Warnf(..., err)		
			}
			if strings.HasPrefix(rel "..") {
				// #1: The path should be joined
				relPath = filepath.Join(base, rel)
			} else {
				// #2: Use the path as is
				relPath = path
			}
			relPath = filepath.ToSlash(relPath)
			relativePaths = append(relativePaths, relPath)
```

Reply:
Done